aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/ParameterGrouper.java

```diff
@@ -0,0 +1,67 @@
+package software.amazon.rds.dbparametergroup;
+
+import software.amazon.awssdk.services.rds.model.Parameter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ParameterGrouper implements AbstractGrouper<String, Parameter> {
+    private void fillPartitionWithIndependantParameters(Map<String, Parameter> parametersToUpdate, List<Parameter> group, int groupSize) {
+        for (String key : parametersToUpdate.keySet()) {
+            if (group.size() == groupSize) {
+                return;
+            }
+            group.add(parametersToUpdate.remove(key));
+        }
+    }
+
+    private void addNewPartitionWithDependantParameters(List<List<Parameter>> partitions, List<List<Parameter>> dependantParameters, int groupSize) {
```

Comment:
Seems tricky to know the `dependantParameters`. Also there is probability that dependent parameters may change for same engine but different versions. So we might ending up filling those based on engine version.